### PR TITLE
Muammer

### PR DESCRIPTION
**Pull request by muammertop**

### CODE REVIEW

File: addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php [hunk 1]
```diff
@@ -2,47 +2,46 @@
...
-use Visiosoft\CatsModule\Category\Contract\CategoryRepositoryInterface;
-use Visiosoft\LocationModule\Country\Contract\CountryRepositoryInterface;
-use Anomaly\Streams\Platform\Message\MessageBag;
 use Visiosoft\PackagesModule\Package\PackageModel;
-
-use Illuminate\Contracts\Events\Dispatcher;
+use Visiosoft\ProfileModule\Adress\Contract\AdressRepositoryInterface;
 use Visiosoft\QrcontactModule\Qr\QrModel;
 use Visiosoft\StoreModule\Ad\AdModel;
+use function PMA\Util\get;
```
Comment (fatihalp) [inline addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php:44]: do we need? @muammertop

File: addons/default/visiosoft/advs-module/resources/views/ad-detail/partials/content.twig [hunk 2]
```diff
@@ -37,7 +37,9 @@
                 </div>
 
                 <!-- Content With Block -->
-                {{ addBlock('ad-detail/content',{'features':features,'adv':adv, 'profile': profile})|raw }}
+                <div id="nav-comments" class="tab-pane fade in">
```
Comment (diashalabi) [inline addons/default/visiosoft/advs-module/resources/views/ad-detail/partials/content.twig:40]: you can't add external module related stuff in the advs module. you can move this to the comment module

File: addons/default/visiosoft/advs-module/resources/views/ad-detail/partials/content.twig [hunk 3]
```diff
@@ -37,7 +37,9 @@
...
+                    {{ addBlock('ad-detail/content',{'features':features,'entry':adv, 'profile': profile, 'commenturl': 'advs::saveComment', 'entry_type': 'Visiosoft\\AdvsModule\\Adv\\AdvModel'})|raw }}
+                </div>
```
Comment (diashalabi) [inline addons/default/visiosoft/advs-module/resources/views/ad-detail/partials/content.twig:42]: You can move the comment url to the comment module. why are you sending the entry_type!?

File: addons/default/visiosoft/advs-module/src/AdvsModuleServiceProvider.php [hunk 4]
```diff
@@ -209,6 +209,11 @@ class AdvsModuleServiceProvider extends AddonServiceProvider
 
         // Others
         'advs/ttr/{id}' => 'Visiosoft\PackagesModule\Http\Controller\packageFEController@advsStatusbyUser',
+
+        'advs/comments/save/{id}' => [
+            'as' => 'advs::saveComment',
+            'uses' => 'Visiosoft\AdvsModule\Http\Controller\AdvsController@saveComment'
+        ]
```
Comment (diashalabi) [inline addons/default/visiosoft/advs-module/src/AdvsModuleServiceProvider.php:216]: you can move it to the comment module

File: addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php [hunk 5]
```diff
@@ -1075,4 +1070,21 @@ public function extendSingle($adId)
         $this->messages->success(trans('visiosoft.module.advs::message.extended', ['number' => $adsExtended]));
         return $this->redirect->back();
     }
+
+    public function saveComment($id)
```
Comment (diashalabi) [inline addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php:1074]: you can move this to the comment module

File: addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php [hunk 6]
```diff
@@ -1075,4 +1070,21 @@ public function extendSingle($adId)
...
+    {
+        $comment = $this->request->validate([
+            'rating' => 'required',
+            'username' => 'required',
+            'title' => 'required',
+            'detail' => 'required',
+        ]);
```
Comment (diashalabi) [inline addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php:1081]: Maybe its better to move the validation logic to a FormRequest class

**Review by diashalabi — CHANGES_REQUESTED** [review]

File: addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php (hunk 5 above)
Comment (fatihalp) [reply addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php:1074]: what we talked :) same thing

File: addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php (hunk 1 above)
Comment (muammertop) [reply addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php:44]: I didn't add it, but I guess it's not working for us.

Comment (fatihalp) [reply addons/default/visiosoft/advs-module/src/Http/Controller/advsController.php:44]: php storm could have been add that. just remove that line

File: addons/default/visiosoft/advs-module/resources/views/ad-detail/partials/content.twig [hunk 7]
```diff
@@ -37,7 +37,8 @@
                 </div>
 
                 <!-- Content With Block -->
-                {{ addBlock('ad-detail/content',{'features':features,'adv':adv, 'profile': profile})|raw }}
+                    {{ addBlock('ad-detail/content',{'features':features,'adv': adv, 'type': 'adv', 'profile': profile,
+                        'commenturl': 'comments::save_comment'})|raw }}
```
Comment (diashalabi) [inline addons/default/visiosoft/advs-module/resources/views/ad-detail/partials/content.twig:41]: please move the commenturl to the comment module

**Review by diashalabi — CHANGES_REQUESTED** [review]

**Review by spektra2147 — APPROVED** [review]

File: addons/default/visiosoft/advs-module/resources/views/ad-detail/partials/content.twig [hunk 8]
```diff
@@ -37,7 +37,7 @@
                 </div>
 
                 <!-- Content With Block -->
-                {{ addBlock('ad-detail/content',{'features':features,'adv':adv, 'profile': profile})|raw }}
+                    {{ addBlock('ad-detail/content',{'features':features,'adv': adv, 'type': 'adv', 'profile': profile})|raw }}
```
Comment (diashalabi) [inline addons/default/visiosoft/advs-module/resources/views/ad-detail/partials/content.twig:40]: Are you using type?

**Review by diashalabi — CHANGES_REQUESTED** [review]

**Review by diashalabi — APPROVED** [review]

**Pull request merged**